multihost_runner.py

```diff
@@ -342,37 +356,34 @@ def main() -> None:
   print("Starting multihost runner...", flush=True)
 
   #### Parse flags ####
-  tpu_prefix = args.TPU_PREFIX
-  script_dir = args.SCRIPT_DIR
-  main_command = args.COMMAND
-  run_name = args.RUN_NAME
-  use_existing_folder = args.USE_EXISTING_FOLDER
-  internal_ip = args.INTERNAL_IP
+  if not args.PROJECT:
+    args.PROJECT = get_project()
```

Comment:
IMO, it's not the best practice to set the args values backward. we can consider setting these as global vairables instead?

Reply:
Changed to set the default in the flag definition. In much older versions of the code we used a different flag parser where I think this was undesirable because the --help printed the default values (which would run these weird functions in this case), but with argparse the default values are not printed with --help, so using a complex function in the default seems fine